torch/onnx/_internal/exporter.py

```diff
@@ -898,17 +956,10 @@ def _assert_fake_tensor_mode(self):
         if (
             has_any_non_fake_tensors or has_any_non_fake_param_or_buffer
         ) and self.options.fake_context:
-            raise OnnxExporterError(
-                self.options.diagnostic_context,
+            raise RuntimeError(
```

Comment:
I didn't follow why the error is changed? Just for my education. Thanks!

Reply:
We are now making `OnnxExporterError` the top level user facing error, encapsulating an `ExportOutput` instance. 

Hence, any error raised from within exporter should raise in its own corresponding error type. Top level export call will catch these and encapsulate accordingly.